### ISSUE TITLE
Custom metric doesn't show up in fit_loader()

### ISSUE THREAD

**Issue by arciel** [achaiah/pywick#5]
I wrote the following custom metric,

```python
class F1Score(pywick.metrics.Metric):
    def __init__(self, micmac='micro'):
        super().__init__()
        self._name = f'f1_{micmac}'
        self.micmac = micmac
        self.correct = []
        self.preds = []

    def reset(self):
        self.correct = []
        self.preds = []

    def __call__(self, inputs, y_pred, y_true, is_val=False):
        self.correct.extend(y_true.detach().to('cpu').numpy().tolist())
        self.preds.extend(y_pred.detach().to('cpu').numpy().argmax(axis=1).tolist())
        f1 = sklearn.metrics.f1_score(self.correct, self.preds, average=self.micmac)
        return f1
```
and use it like so

```
metrics = [F1Score('micro'), CategoricalAccuracySingleInput(top_k=1)]
trainer.compile(optimizer='adam',
                criterion='cross_entropy',
                initializers=initializers,
                metrics=metrics)
```

But while running `trainer.fit_loader(...)`, I get the following output
```
Epoch 1/1:   7%|▋         | 7/98 [00:12<02:46,  1.83s/ batches, loss=0.6508, top_1:acc=76.04, learn_rates=[0.001]]
```
No mention of the custom metric anywhere!

But running `trainer.eval_loader()` gives,
```
{'val_f1_micro': 0.7514395393474088,
 'val_loss': 0.6613252784975586,
 'val_top_1:acc_metric': 75.14395393474088}
```
Which works fine, so what's going on with `fit_loader()`?

**Comment by achaiah** [comment]
Can you please verify that your metric is returning a value and not "None" during the training phase? It should be automatically incorporated into the logs unless there is no value returned when the __call__(...) method executes.

This is the code that handles metric logs:

```
        for metric in self.metrics:
            metric_out = metric(input_batch, output_batch, target_batch, is_val)
            if metric_out is not None:
                logs[self.prefix + metric._name] = metric_out
        return logs
```
Since you see output during the validation phase the metric is correctly added to the list and I can't think of many reasons why it shouldn't show up.
